frontend/console/src/features/modules/decls/DeclLink.tsx

```diff
@@ -1,33 +1,48 @@
-import { useMemo, useRef, useState } from 'react'
+import { useEffect, useMemo, useRef, useState } from 'react'
 import { useNavigate } from 'react-router-dom'
-import { useModules } from '../../../api/modules/use-modules'
+import { useStreamModules } from '../../../api/modules/use-stream-modules'
 import { classNames } from '../../../utils'
+import { getTreeWidthFromLS } from '../module.utils'
 import { Schema } from '../schema/Schema'
-import { type DeclSchema, declFromModules } from '../schema/schema.utils'
+import { type DeclSchema, declSchemaFromModules } from '../schema/schema.utils'
+
+const SnippetContainer = ({
+  decl,
+  moduleName,
+  visible,
+  linkRect,
+  containerRect,
+}: { decl: DeclSchema; moduleName: string; visible: boolean; linkRect?: DOMRect; containerRect?: DOMRect }) => {
+  const [containerX, setContainerX] = useState(containerRect?.x || getTreeWidthFromLS())
+  useEffect(() => {
+    if (containerRect !== undefined) {
+      setContainerX(containerRect.x)
+    }
+  }, [containerRect])
 
-const SnippetContainer = ({ decl, linkRect, containerRect }: { decl: DeclSchema; linkRect?: DOMRect; containerRect?: DOMRect }) => {
   const ref = useRef<HTMLDivElement>(null)
   const snipRect = ref?.current?.getBoundingClientRect()
 
   const hasRects = !!snipRect && !!linkRect
-  const toTop = hasRects && window.innerHeight - linkRect.top - linkRect.height < snipRect.height + linkRect.height
+  const fitsAbove = hasRects && linkRect.top - 64 > snipRect.height
```

Comment:
What is the `64` here?

Reply:
Oooo whoops thanks for catching that! This is the height of the topbar. I'll move it to a const. Completely slipped my mind